cmd/distribution.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2020 SIGHUP s.r.l All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+
+	getter "github.com/hashicorp/go-getter"
+	"github.com/spf13/cobra"
+)
+
+const (
+	furyFile                    = "Furyfile.yml"
+	kustomizationFile           = "kustomization.yaml"
+	httpsDistributionRepoPrefix = "http::https://github.com/sighupio/fury-distribution/releases/download/"
+)
+
+var (
+    distributionVersion string
+	initKustomize bool
+
+	distributionCmd = &cobra.Command{
+		Use:   "distribution",
+		Short: "Manages Kubernetes Fury Distribution, initialize Furyfile.yml and download Fury distribution modules",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			err = cmd.Help()
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	initCmd = &cobra.Command{
+		Use:   "init",
+		Short: "Initialize the minimum distribution configuration",
+		Long:  "Initialize the current directory with the minimum distribution configuration",
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			// if distributionVersion is empty throw error
+			if distributionVersion == "" {
+				return fmt.Errorf("--version <version> flag is required")
+			}
+
+			url := httpsDistributionRepoPrefix + distributionVersion + "/" + furyFile
+			err = downloadFile(url, furyFile)
+			if err != nil {
+				return err
+			}
+
+			if initKustomize {
+				url := httpsDistributionRepoPrefix + distributionVersion + "/" + kustomizationFile
+				err = downloadFile(url, kustomizationFile)
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		},
+	}
+	downloadCmd = &cobra.Command{
+		Use:   "download",
+		Short: "Download dependencies specified in Furyfile.yml",
+		Long:  "Download dependencies specified in Furyfile.yml",
+		Run: func(cmd *cobra.Command, args []string) {
+			viper.SetConfigType("yml")
+			viper.AddConfigPath(".")
+			viper.SetConfigName(configFile)
+			config := new(Furyconf)
+			if err := viper.ReadInConfig(); err != nil {
+				logrus.Fatalf("Error reading config file, %s", err)
+			}
+			err := viper.Unmarshal(config)
+			if err != nil {
+				logrus.Fatalf("unable to decode into struct, %v", err)
```

Comment:
can we improve this error message to make it more user-friendly?

Reply:
Yes, but it's something I did not write, i inherited from the refactoring